### PR TITLE
[16.0][ADD] stock_picking_auto_create_lot_sequence

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This module acts as a bridge between the `stock_picking_auto_create_lot` and `product_lot_sequence` modules.
Due to the changes introduced in [this commit](https://github.com/OCA/stock-logistics-workflow/commit/1f1d1cd4ba102f35971c074ca869ed6f7113cd2b) for stock_picking_auto_create_lot, using these two modules together disrupts the design of `product_lot_sequence` by not allowing the use of specific sequences for each product. This module restores the intended behavior.
Related: https://github.com/OCA/stock-logistics-workflow/pull/1616

@qrtl QT4624

### CODE REVIEW

**Comment by AungKoKoLin1997** [comment]
I created the PR to fix the tests failed.
https://github.com/OCA/stock-logistics-workflow/pull/1621

File: stock_picking_auto_create_lot_sequence/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2024 Quartile Limited
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/__manifest__.py]:
```suggestion
# Copyright 2024 Quartile (https://www.quartile.co)
```

File: stock_picking_auto_create_lot_sequence/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Stock Picking Auto Create Lot Sequence",
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "Quartile Limited, Odoo Community Association (OCA)",
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/__manifest__.py]:
```suggestion
    "author": "Quartile, Odoo Community Association (OCA)",
```

File: stock_picking_auto_create_lot_sequence/models/stock_move_line.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    def _get_lot_sequence(self):
+        self.ensure_one()
+        seq_policy = self.env["stock.lot"]._get_sequence_policy()
+        if (
+            seq_policy == "product"
+            and self.product_id
+            and self.product_id.product_tmpl_id.lot_sequence_id
+        ):
+            return self.product_id.product_tmpl_id.lot_sequence_id._next()
+        return super()._get_lot_sequence()
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/models/stock_move_line.py]:
```suggestion
        if not self.product_id.lot_sequence_id:
            return super()._get_lot_sequence()
        seq_policy = self.env["stock.lot"]._get_sequence_policy()
        if seq_policy != "product":
            return super()._get_lot_sequence()
        return self.product_id.lot_sequence_id._next()
``````

File: stock_picking_auto_create_lot_sequence/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+This module acts as a bridge between the stock_picking_auto_create_lot and product_lot_sequence modules.
+Due to the changes introduced in this commit(https://github.com/OCA/stock-logistics-workflow/commit/1f1d1cd4ba102f35971c074ca869ed6f7113cd2b)
+for stock_picking_auto_create_lot, using these two modules together disrupts the design of product_lot_sequence by not allowing the use of specific
+sequences for each product. This module restores the intended behavior.
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/readme/DESCRIPTION.rst]: This explanation sounds a bit off. I think we can simply remove this part.
```suggestion
```

File: stock_picking_auto_create_lot_sequence/tests/test_stock_picking_auto_create_lot_sequence.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestStockPickingProductLotSequence(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.supplier_location_id = cls.env.ref("stock.stock_location_suppliers").id
+        cls.stock_location_id = cls.env.ref("stock.stock_location_stock").id
+        cls.picking_type_in_id = cls.env.ref("stock.picking_type_in").id
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/tests/test_stock_picking_auto_create_lot_sequence.py]: Might as well do this in the setup.
```suggestion
        picking_type_in = cls.env.ref("stock.picking_type_in")
        picking_type_in.auto_create_lot = True
```

File: stock_picking_auto_create_lot_sequence/tests/test_stock_picking_auto_create_lot_sequence.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                "product_uom_qty": 10,
+                "product_uom": cls.product.uom_id.id,
+                "picking_id": cls.picking.id,
+                "location_id": cls.supplier_location_id,
+                "location_dest_id": cls.stock_location_id,
+            }
+        )
+
+    def test_stock_picking_product_lot_sequence(self):
+        self.picking.picking_type_id.auto_create_lot = True
```
Comment (yostashiro) [inline stock_picking_auto_create_lot_sequence/tests/test_stock_picking_auto_create_lot_sequence.py]:
```suggestion
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Review by yostashiro — APPROVED** [review]
Code review.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Can you please check the CI errors?

**Comment by AungKoKoLin1997** [comment]
> @AungKoKoLin1997 Can you please check the CI errors?

@yostashiro I created the PR for this https://github.com/OCA/stock-logistics-workflow/pull/1619#issuecomment-2160152486.

**Review by kanda999 — APPROVED** [review]
Functional and Code review.

**Comment by rousseldenis** [comment]
/ocabot merge nobump

Should be great to improve coverage by adding tests that shows other scenarios are working

**Pull request merged**